src/plugins/mer/merdeploysteps.cpp

```diff
@@ -223,9 +225,17 @@ bool MerProcessStep::init()
 
 bool MerProcessStep::init(InitOptions options)
 {
-    QmakeBuildConfiguration *bc = qobject_cast<QmakeBuildConfiguration*>(buildConfiguration());
-    if (!bc)
-        bc =  qobject_cast<QmakeBuildConfiguration*>(target()->activeBuildConfiguration());
+    BuildConfiguration *bc = nullptr;
+    if (qobject_cast<CMakeBuildConfiguration*>(buildConfiguration())) {
+        bc = qobject_cast<CMakeBuildConfiguration*>(buildConfiguration());
+        if (!bc)
+            bc =  qobject_cast<CMakeBuildConfiguration*>(target()->activeBuildConfiguration());
+    } else {
+        // Default to QmakeBuildConfiguration
+        bc = qobject_cast<QmakeBuildConfiguration*>(buildConfiguration());
+        if (!bc)
+            bc =  qobject_cast<QmakeBuildConfiguration*>(target()->activeBuildConfiguration());
+    }
```

Comment:
This line is unreachable I think:

`bc =  qobject_cast<CMakeBuildConfiguration*>(target()->activeBuildConfiguration());`

It would be also wort to think if resorting to `target()->activeBuildConfiguration()` is really needed here as `buildConfiguration` already returns that as a fallback, it just doesn't do the casting. Maybe this changed at some point and we can simplify it now here.